src/Microsoft.OData.Edm/ExtensionMethods/TargetPathHelper.cs

```diff
@@ -0,0 +1,287 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TargetPathHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.OData.Edm.Vocabularies;
+
+    /// <summary>
+    /// Target path helper.
+    /// </summary>
+    internal static class TargetPathHelper
+    {
+        /// <summary>
+        /// Get the Segments from a target path string as enumeration of Edm elements.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="targetSegments">Segments for the target path..</param>
+        /// <param name="caseInsensitive">Property name case-insensitive or not.</param>
+        /// <returns>The created enumeration of <see cref="IEdmElement"/>.</returns>
+        public static IEnumerable<IEdmElement> GetTargetSegments(this IEdmModel model, string[] targetSegments, bool caseInsensitive)
+        {
+            EdmUtil.CheckArgumentNull(model, nameof(model));
+            EdmUtil.CheckArgumentNull(targetSegments, nameof(targetSegments));
+
+            IEdmEntityContainer container = model.FindEntityContainer(targetSegments[0]);
+
+            if (container == null)
+            {
+                return Enumerable.Empty<IEdmElement>();
+            }
+
+            List<IEdmElement> pathSegments = new List<IEdmElement>();

```

Comment:
We can initialize the list capacity based on the size of the segments array to reduce the chances of List resizing.